client/src/templates/Challenges/video/show.tsx

```diff
@@ -157,25 +165,45 @@ class ShowVideo extends Component<ShowVideoProps, ShowVideoState> {
     }
   }
 
-  handleSubmit(solution: number, openCompletionModal: () => void) {
-    if (solution - 1 === this.state.selectedOption) {
-      this.setState({
-        showWrong: false
-      });
+  handleSubmit = () => {
+    const {
+      data: {
+        challengeNode: {
+          challenge: { questions }
+        }
+      },
+      openCompletionModal
+    } = this.props;
+
+    // subract 1 because the solutions are 1-indexed
+    const mcqSolutions = questions.map(question => question.solution - 1);
+
+    this.setState({
+      submittedMcqAnswers: this.state.selectedMcqOptions,
+      showFeedback: true
+    });
+
+    const allMcqAnswersCorrect = arraysAreEqual(
+      mcqSolutions,
+      this.state.selectedMcqOptions
+    );
+
+    if (allMcqAnswersCorrect) {
       openCompletionModal();
-    } else {
-      this.setState({
-        showWrong: true
-      });
     }
-  }
+  };
 
-  handleOptionChange = (
-    changeEvent: React.ChangeEvent<HTMLInputElement>
+  handleMcqOptionChange = (
+    questionIndex: number,
+    answerIndex: number
   ): void => {
+    const { selectedMcqOptions } = this.state;
+
+    const newSelectedMcqOptions = [...selectedMcqOptions];
+    newSelectedMcqOptions[questionIndex] = answerIndex;
+
     this.setState({
-      showWrong: false,
-      selectedOption: parseInt(changeEvent.target.value, 10)
+      selectedMcqOptions: newSelectedMcqOptions
```

Comment:
Again, I'd use a map and we need to use the `(state) =>...` form.

Reply:
✅